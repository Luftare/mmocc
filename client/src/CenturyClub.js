import React, { Component } from 'react';
import Button from './Button';
import Time from './Time';

class CenturyClub extends Component {
  constructor(props) {
    super(props);
    this.state = {
      intervalId: null,
      remainingTime: 6000000,
      startTime: 0
    };
  }

  timerIsActive = () => {
    return !(this.state.intervalId == null);
  };

  handleStartStopButtonClick = () => {
    if (this.timerIsActive()) {
      this.stopTimer();
    } else {
      this.startTimer();
    }
  };

  startTimer = () => {
    const currentTime = Date.now();
    const intervalId = setInterval(this.loop, 100);
    this.setState({ intervalId, startTime: currentTime });
  };

  stopTimer = () => {
    clearInterval(this.state.intervalId);
    this.setState({ intervalId: null });
  };

  loop = () => {
    const currentTime = Date.now();
    const passedTime = currentTime - this.state.startTime;
    const remainingTime = 6000000 - passedTime;
    this.setState({ remainingTime });
  };

  render() {
    const buttonText = this.timerIsActive() ? 'Give up' : 'Start';
    return (
      <div>
        <h1>shiii</h1>
        <Button text={buttonText} onClick={this.handleStartStopButtonClick} />
        <Time milliseconds={this.state.remainingTime} />
      </div>
    );
  }
}

export default CenturyClub;
