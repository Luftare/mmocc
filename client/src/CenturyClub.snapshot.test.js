import React from 'react';
import CenturyClub from './CenturyClub';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

import { configure, mount } from 'enzyme';

configure({ adapter: new Adapter() });

describe('CenturyClub snapshot', () => {
  it('renders correnctly', () => {
    const centuryClub = renderer.create(<CenturyClub />);
    expect(centuryClub).toMatchSnapshot();
  });
});
