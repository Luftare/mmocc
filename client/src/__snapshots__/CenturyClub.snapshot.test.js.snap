// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CenturyClub snapshot renders correnctly 1`] = `
<div>
  <h1>
    shiii
  </h1>
  <button
    onClick={[Function]}
  >
    Start
  </button>
  <div>
    100
     min 
    0
     sec
  </div>
  <span>
    100
     %
  </span>
  <span>
    100
     %
  </span>
  <span>
    100
     %
  </span>
  <span>
    100
     %
  </span>
  <span>
    100
     %
  </span>
  <span>
    100
     %
  </span>
  <span>
    100
     %
  </span>
  <span>
    100
     %
  </span>
  <span>
    100
     %
  </span>
  <span>
    100
     %
  </span>
  <span>
    100
     %
  </span>
  <span>
    100
     %
  </span>
</div>
`;
